Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -6246,14 +6246,15 @@ void WebPage::didChangeSelection(Frame& frame)
     didChangeSelectionOrOverflowScrollPosition();
 
 #if PLATFORM(IOS_FAMILY)
-    if (!m_sendAutocorrectionContextAfterFocusingElement)
+    if (!std::exchange(m_sendAutocorrectionContextAfterFocusingElement, false))
         return;
 
-    if (UNLIKELY(!frame.document() || !frame.document()->hasLivingRenderTree() || frame.selection().isNone()))
-        return;
+    callOnMainRunLoop([protectedThis = Ref { *this }, frame = Ref { frame }] {
```

Comment:
could/should we optimize this such that we don't wait if we're not in the middle of layout?

Reply:
Hm...I don't think so. It's not about whether or not layout is up to date at this point, but rather about whether or not we do anything that might _require_ layout to be up to date.